src/charm.py

```diff
@@ -41,83 +65,51 @@ def __init__(self, *args):
 
     def _update_prometheus_jobs(self, _):
         """Setup Prometheus scrape configuration for external targets."""
-        self.unit.status = ActiveStatus()
-        if not self.unit.is_leader():
-            return
-
-        jobs = self._scrape_jobs()
-        if not jobs:
-            return
-
-        for relation in self.model.relations[self._prometheus_relation]:
-            relation.data[self.app]["scrape_jobs"] = json.dumps(jobs)
+        if self.unit.is_leader():
+            if jobs := self._scrape_jobs():
+                for relation in self.model.relations[self._prometheus_relation]:
+                    relation.data[self.app]["scrape_jobs"] = json.dumps(jobs)
 
-    def _scrape_jobs(self):
-        targets = self._targets()
+                self.unit.status = ActiveStatus()
 
-        jobs = (
-            [
+    def _scrape_jobs(self) -> list:
+        if targets := self._targets():
+            labels = self._labels()
```

Comment:
There was good reason why labels were not always being added. It leads to uglyness such as labels `labels: {}`. Please revert this change.